c++/src/kj/async-io-unix.c++

```diff
@@ -1101,19 +1101,21 @@ public:
         // Ignore disallowed address.
         return acceptImpl(authenticated);
       } else {
-        // TODO(perf):  As a hack for the 0.4 release we are always setting
-        //   TCP_NODELAY because Nagle's algorithm pretty much kills Cap'n Proto's
-        //   RPC protocol.  Later, we should extend the interface to provide more
-        //   control over this.  Perhaps write() should have a flag which
-        //   specifies whether to pass MSG_MORE.
-        int one = 1;
-        KJ_SYSCALL_HANDLE_ERRORS(::setsockopt(
-              ownFd.get(), IPPROTO_TCP, TCP_NODELAY, (char*)&one, sizeof(one))) {
-          case EOPNOTSUPP:
-            break;
-          default:
-            KJ_FAIL_SYSCALL("setsocketopt(IPPROTO_TCP, TCP_NODELAY)", error);
-        }
+	if (addr.ss_family != AF_UNIX) {
+	  // TODO(perf):  As a hack for the 0.4 release we are always setting
+	  //   TCP_NODELAY because Nagle's algorithm pretty much kills Cap'n Proto's
+	  //   RPC protocol.  Later, we should extend the interface to provide more
+	  //   control over this.  Perhaps write() should have a flag which
+	  //   specifies whether to pass MSG_MORE.
+	  int one = 1;
+	  KJ_SYSCALL_HANDLE_ERRORS(::setsockopt(
+		ownFd.get(), IPPROTO_TCP, TCP_NODELAY, (char*)&one, sizeof(one))) {
+	    case EOPNOTSUPP:
```

Comment:
Oh, I guess another possible explanation for why the tests were passing on non-cygwin is that implementations return EOPNOTSUPP for unix domain sockets?  In which case, the cygwin error message ("Protocol not available") suggests that it might have been returning EPROTONOSUPPORT instead.  So an alternative solution would be to just add that to the list of ignored cases.

Reply:
Yeah that's probably a better solution, in case anyone ever uses this code with families other than IP and Unix.